### ISSUE TITLE
io.load_delimited fails for unicode strings

### ISSUE THREAD

**Issue by rabitt** [craffel/mir_eval#117]
It's easy to work around this problem, but I'll mention it regardless. When a file path that is passed to io.load_delimited is a unicode string, a value error is raised.

```
$ nonexistent_file = u"fake_filepath.csv"
$ mir_eval.io.load_delimited(nonexistent_file, ',')
ValueError: filename must be a string or file handle
```

**Comment by craffel** [comment]
I didn't even know filesystems supported unicode filenames.  The fix would be to change https://github.com/craffel/mir_eval/blob/develop/mir_eval/input_output.py#L54 to 
```
    if type(filename) == str or type(filename) == unicode:
```

**Comment by bmcfee** [comment]
> if type(filename) == str or type(filename) == unicode:

Two things:

1) use `isinstance`, not type equivalence.  The code above won't handle subclasses of str or unicode, but you probably should support that.

2) `six` provides a sane version of this through `six.stringtypes`.  see [here](https://pythonhosted.org/six/)

**Comment by craffel** [comment]
Ok, if/when we add `six` as a dependency (cc https://github.com/craffel/mir_eval/pull/113#discussion_r26895157 ) we should do `isinstance(six.stringtypes)`.

**Comment by bmcfee** [comment]
Even if you're really dead-set against adding dependencies, six is intentionally self-contained as a single file that you can drop into your codebase.  Scipy does this (see: scipy/_lib/six.py).

**Comment by craffel** [comment]
I'm totally fine with adding `six` as a dependency.

**Comment by urinieto** [comment]
Full support to the idea of adding six as a dependency.

**Comment by craffel** [comment]
Addressed in #135
